base/interactiveutil.jl

```diff
@@ -409,3 +409,258 @@ function runtests(tests = ["all"], numcores = ceil(Int,CPU_CORES/2))
+            end
+            line = resize!(line, newline)
+            line = read!(head, line)
+
+            write(io, line)
+            if nb_available(head) > 0 # more to read? replace with ...
+                print(io, '\u2026') # hdots
+            end
+            println(io)
+            seekend(head) # skip the rest of the text
+        end
+    end
+end
+whos(m::Module, pat::Regex=r"") = whos(STDOUT, m, pat)
+whos(pat::Regex) = whos(STDOUT, current_module(), pat)
+
+"""
+    summarysize(obj, recurse) => Int
+
+summarysize is an estimate of the size of the object
+as if all iterables were allocated inline
+in general, this forms a conservative lower bound
+n the memory "controlled" by the object
+if recurse is true, then simply reachable memory
+should also be included, otherwise, only
+directly used memory should be included
+you should never ignore recurse in cases where recursion is possible"""
+summarysize(obj::ANY, recurse::Bool) = try convert(Int, sizeof(obj)); catch; Core.sizeof(obj); end
+
+# these three cases override the exception that would be thrown by Core.sizeof
+summarysize(obj::Symbol, recurse::Bool) = 0
+summarysize(obj::DataType, recurse::Bool) = 0
+function summarysize(obj::Module, recurse::Bool)
+    size::Int = sizeof(obj)
+    if recurse
+        for binding in names(obj, true)
+            if isdefined(obj, binding)
+                value = getfield(obj, binding)
+                if (value !== obj) # skip the self-recursive definition
+                    recurseok = !isa(value, Module) || module_parent(value) === obj
+                    size += summarysize(value, recurseok)::Int # recurse on anything that isn't a module
+                end
+            end
+        end
+    end
+    return size
+end
+
+function summarysize(obj::Task, recurse::Bool)
+    size::Int = sizeof(obj)
+    if recurse
+        if isdefined(obj, :code)
+            size += summarysize(obj.code, true)::Int
+        end
+        size += summarysize(obj.storage, true)::Int
+
+        size += summarysize(obj.backtrace, false)::Int
+        size += summarysize(obj.donenotify, false)::Int
+        size += summarysize(obj.exception, false)::Int
+        size += summarysize(obj.result, false)::Int
+    end
+    return size
+end
+
+function summarysize(obj::SimpleVector, recurse::Bool)
+    size::Int = sizeof(obj)
+    if recurse
+        for val in obj
+            if val !== obj
+                size += summarysize(val, false)::Int
+            end
+        end
+    end
+    return size
+end
+
+function summarysize(obj::Tuple, recurse::Bool)
+    size::Int = sizeof(obj)
+    if recurse
+        for val in obj
+            if val !== obj && !isbits(val)
+                size += summarysize(val, recurse)::Int
+            end
+        end
+    end
+    return size
+end
+
+function summarysize(obj::Array, recurse::Bool)
+    size::Int = sizeof(obj)
+    if recurse && !isbits(eltype(obj))
+        for i in 1:length(obj)
```

Comment:
Isn't `eachindex` just as fast as `1:length(obj)`?  Seems if so, be consistent and always use `eachindex` (or renamed to `indexes`, as discussed elsewhere?)